packages/excalidraw/zindex.ts

```diff
@@ -383,26 +378,27 @@ const shiftElementsToEnd = (
     }
   }
 
-  const targetElements = Object.values(targetElementsMap).map((element) => {
-    return bumpVersion(element);
-  });
+  const targetElements = Array.from(targetElementsMap.values());
 
   const leadingElements = elements.slice(0, leadingIndex);
   const trailingElements = elements.slice(trailingIndex + 1);
 
-  return direction === "left"
-    ? [
-        ...leadingElements,
-        ...targetElements,
-        ...displacedElements,
-        ...trailingElements,
-      ]
-    : [
-        ...leadingElements,
-        ...displacedElements,
-        ...targetElements,
-        ...trailingElements,
-      ];
+  return updateFractionalIndices(
+    direction === "left"
+      ? [
+          ...leadingElements,
+          ...targetElements,
+          ...displacedElements,
+          ...trailingElements,
+        ]
+      : [
+          ...leadingElements,
+          ...displacedElements,
+          ...targetElements,
+          ...trailingElements,
+        ],
+    targetElementsMap,
```

Comment:
We do generate a new index in these z-index actions even though the order didn't change (i.e. bringing the item forward when it's already on top). Would not put this logic into fractional indices though, instead `targetElementsMap` should not contain elements which were not really re-ordered.

cc @dwelle @ryan-di

Reply:
Will try to fix it - hopefully an easy one, afterwards we could merge this.